### DISCUSSION TITLE
IDL solution

### DISCUSSION THREAD

**Discussion by jwinarske** [fzyzcjy/flutter_rust_bridge#770] [Q&A] [answered]
What's the suggested pattern when using with IDL such as protobuf?

**Comment by fzyzcjy** [comment, marked as answer]
Maybe serialize it to `Vec<u8>` (Uint8LIst)?

**Reply by jwinarske** [reply]
Using https://pub.dev/packages/pbconv on the Dart side.

**Reply by fzyzcjy** [reply]
btw seems https://pub.dev/packages/protobuf is more widely used

**Comment by jwinarske** [comment]
Say I want to in the same project use gRPC.  I include Tonic, and a proto file.  Build.rs is used to generate Rust code from protobuf.  How can I programmatically decorate the struct/enums/etc in the intermediate IDL output?  I don't want to check in the intermediate code.

Example struct definition in intermediate code:

```
#[derive(Clone, PartialEq, ::prost::Message)]
pub struct Datapoint {
    /// Timestamp of the value
    #[prost(message, optional, tag = "1")]
    pub timestamp: ::core::option::Option<::prost_types::Timestamp>,
    /// values
    #[prost(
        oneof = "datapoint::Value",
        tags = "10, 11, 12, 13, 14, 15, 16, 17, 18, 21, 22, 23, 24, 25, 26, 27, 28"
    )]
    pub value: ::core::option::Option<datapoint::Value>,
}
```

**Reply by fzyzcjy** [reply]
I guess that looks like a question for the protobuf codegen lib?

**Reply by jwinarske** [reply]
I'm thinking a macro helper might be in order.

**Reply by fzyzcjy** [reply]
But, if you have a protobuf, why not just serialize it and send the serialized bytes via bridge

**Reply by jwinarske** [reply]
I'm looking for a zero copy solution.
